### PR TITLE
Fix/broken vrt tests

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #2172

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
     "watch": "npm run build:dev -- --watch",
     "release-zip": "npm run build && npm run zip",
     "test": "npm run build:production -- --display=none && bundlesize",
-    "test:visualtest": "npm run build:static && start-server-and-test storybook:vrt http-get://localhost:9001 backstopjs",
+    "pretest:visualtest": "npm run build:storybook --silent",
+    "test:visualtest": "npm run backstopjs -- --storybook-host=./dist/",
```
Comment (aaemnnosttv) [inline package.json:22]: Passing `--storybook-host` here results in the argument being set twice in the `config.js`

```
[ '/usr/local/bin/node',
  '/usr/local/bin/backstop',
  'test',
  '--config=tests/backstop/config.js',
  '--moby',
  '--storybook-host=./dist/',
  '--storybook-host=http://host.docker.internal:9001/' ]
```

File: tests/backstop/config.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const viewports = require( './viewports' );
 // This will be passed through with the `backstop` command run with docker.
 if ( process.argv.includes( '--docker' ) ) {
 	const hostname = require( './detect-storybook-host' );
-	process.argv.push( `--storybook-host=${ hostname }` );
+	process.argv.push( `--storybook-host=http://${ hostname }:9001/` );
```
Comment (aaemnnosttv) [inline tests/backstop/config.js:26]: Why was this changed?

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
This is great, just a few questions and suggestions.

File: tests/backstop/config.js (hunk 2 above)
Comment (eugene-manuilov) [reply tests/backstop/config.js:26]: It has been moved from the `scenarios.js` file to not break URLs when `./dist/` is used as a host.

File: package.json [hunk 3]
```diff
@@ -18,7 +18,8 @@
     "watch": "npm run build:dev -- --watch",
     "release-zip": "npm run build && npm run zip",
     "test": "npm run build:production -- --display=none && bundlesize",
-    "test:visualtest": "npm run build:static && start-server-and-test storybook:vrt http-get://localhost:9001 backstopjs",
+    "pretest:visualtest": "npm run build:storybook --silent",
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply package.json]: Updated.

File: tests/backstop/config.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ module.exports = {
 	debugWindow: false,
 	// Use a custom command template to make sure it works correctly in the GitHub actions environment.
 	// The only difference between the original dockerCommandTemplate and this one is that there is no -t flag in the current template.
-	dockerCommandTemplate: 'docker run --rm -i --mount type=bind,source="{cwd}",target=/src backstopjs/backstopjs:{version} {backstopCommand} {args}',
+	dockerCommandTemplate: `docker run --rm -i${ process.stdout.isTTY ? 't' : '' } --mount type=bind,source="{cwd}",target=/src backstopjs/backstopjs:{version} {backstopCommand} {args}`,
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply tests/backstop/config.js]: Updated.

File: package.json (hunk 1 above)
Comment (eugene-manuilov) [reply package.json:23]: Updated.

File: package.json [hunk 5]
```diff
@@ -9,7 +9,8 @@
     "build:production": "npm run remove-dist && npm run build:static && webpack --mode=production",
     "build:test": "npm run build:production -- --include-tests",
     "build:static": "gulp imagemin",
-    "build:storybook": "build-storybook -c .storybook -o dist && npm run build:static",
+    "build:storybook": "build-storybook -c .storybook -o dist",
+    "postbuild:storybook": "npm run build:static --silent",
```
Comment (aaemnnosttv) [inline package.json:13]: This requires the `--` to work.
```suggestion
    "postbuild:storybook": "npm run build:static -- --silent",
```

File: package.json [hunk 6]
```diff
@@ -9,7 +9,8 @@
     "build:production": "npm run remove-dist && npm run build:static && webpack --mode=production",
     "build:test": "npm run build:production -- --include-tests",
     "build:static": "gulp imagemin",
-    "build:storybook": "build-storybook -c .storybook -o dist && npm run build:static",
+    "build:storybook": "build-storybook -c .storybook -o dist",
```
Comment (aaemnnosttv) [inline package.json:12]: Let's put these here as previously suggested so that both the build-job and the VRT job benefit from it.
```suggestion
    "build:storybook": "build-storybook -c .storybook -o dist --quiet --loglevel warn",
```

File: package.json [hunk 7]
```diff
@@ -18,7 +19,8 @@
     "watch": "npm run build:dev -- --watch",
     "release-zip": "npm run build && npm run zip",
     "test": "npm run build:production -- --display=none && bundlesize",
-    "test:visualtest": "npm run build:static && start-server-and-test storybook:vrt http-get://localhost:9001 backstopjs",
+    "pretest:visualtest": "npm run build:storybook --silent -- --quiet --loglevel warn",
```
Comment (aaemnnosttv) [inline package.json:22]: Then these are no longer needed here.
```suggestion
    "pretest:visualtest": "npm run build:storybook",
```

**Review by aaemnnosttv — COMMENTED** [review]
Looks great, just these last few tweaks which I will apply.

**Review by aaemnnosttv — APPROVED** [review]
Amazin'! 🎉
